src/kafka/base.rs

```diff
@@ -163,8 +163,31 @@ pub trait AlertProducer {
             )?;
             debug!(?total_messages);
 
-            // TODO: need to get the expected count
-            // TODO: finish the produce-or-not logic
+            // Count the number of Avro files in the data directory
+            let avro_count = count_files_in_dir(&self.data_directory(), Some(&["avro"]))?;
+
+            // If they don't match, delete all messages from the topic and
+            // start fresh
+            if total_messages > 0 && total_messages != avro_count as u32 {
```

Comment:
```suggestion
            if total_messages != avro_count as u32 {
```

Reply:
I started with this logic, but ended up wanting to handle the case where we had zero messages, which means we can go ahead and send them in. Am I missing the fact that the size of the topic can't be zero?

Reply:
I'm looking at it this way: only skip producing if the message count equals the avro count; in all other cases, delete the topic if it exists and then download+produce. To me, it shouldn't be a big deal to delete an empty topic (though maybe I'm overlooking a reason why that's a bad idea).

The fact that deleting a topic is a TODO item here shouldn't matter for us in practice. Really, having the message count not equal the avro file count is a pretty weird edge case. I can't think of any normal workflow that would regularly leave one of us with an empty topic or a topic whose message count is more/less than it should be. And when that does happen, we're already used to dropping the topic manually.

Reply:
> Hmmmm that doesn't sound right to me. If we have more than the expected number of messages for benchmarking our runs won't be consistent, since it does take some processing time to figure out that we already have an alert to avoid reprocessing it fully.

Ah you're right. I retract my last comment that overproducing would be ok for benchmarking.